subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_common.c

```diff
@@ -16,6 +16,89 @@
 
 LOG_MODULE_REGISTER(nrf_cloud_fota_common, CONFIG_NRF_CLOUD_LOG_LEVEL);
 
+enum nrf_cloud_fota_validate_status boot_fota_validate_get(
+	const enum nrf_cloud_fota_bootloader_status_flags bl_flags)
+{
+	enum nrf_cloud_fota_validate_status validate = NRF_CLOUD_FOTA_VALIDATE_UNKNOWN;
+
+	/* Rebooted, compare active slot with previous, if set */
+	if (bl_flags & NRF_CLOUD_FOTA_BL_STATUS_S0_FLAG_SET) {
+#if defined(CONFIG_FOTA_DOWNLOAD)
+		/* If the slot has changed, so has the (b1) bootloader */
+		bool s0_active;
+		bool s0_prev = bl_flags & NRF_CLOUD_FOTA_BL_STATUS_S0_WAS_ACTIVE;
+		int err = fota_download_s0_active_get(&s0_active);
+
+		if (err) {
+			LOG_WRN("Active slot unknown, error: %d", err);
+		} else if (s0_active != s0_prev) {
+			LOG_INF("Bootloader slot changed, FOTA update validated");
+			validate = NRF_CLOUD_FOTA_VALIDATE_PASS;
+		} else {
+			LOG_WRN("Bootloader slot unchanged, FOTA update invalidated");
+			validate = NRF_CLOUD_FOTA_VALIDATE_FAIL;
+		}
+#endif
+	}
+
+	if (validate == NRF_CLOUD_FOTA_VALIDATE_UNKNOWN) {
+		LOG_WRN("Bootloader FOTA update complete but not validated");
+	}
+
+	return validate;
+}
+
+enum nrf_cloud_fota_validate_status app_fota_validate_get(void)
+{
+	enum nrf_cloud_fota_validate_status validate = NRF_CLOUD_FOTA_VALIDATE_UNKNOWN;
+
+#if defined(CONFIG_MCUBOOT_IMG_MANAGER)
+	if (!boot_is_img_confirmed()) {
+		int err = boot_write_img_confirmed();
+
+		if (err) {
+			LOG_ERR("Application FOTA update confirmation failed: %d", err);
+			/* If this fails then MCUBOOT will revert
+			* to the previous image on reboot
+			*/
+			validate = NRF_CLOUD_FOTA_VALIDATE_FAIL;
+			LOG_INF("Previous version will be restored on next boot");
+		} else {
+			LOG_INF("App FOTA update confirmed");
+			validate = NRF_CLOUD_FOTA_VALIDATE_PASS;
+		}
+	}
+#endif
+
+	return validate;
+}
+
+enum nrf_cloud_fota_validate_status modem_fota_validate_get(void)
+{
+#if defined(CONFIG_NRF_MODEM_LIB)
+	int modem_lib_init_result = nrf_modem_lib_get_init_ret();
```

Comment:
fyi, I am planning to deprecate this function in favor of receiving a callback upon libmodem's initialization via the `NRF_MODEM_LIB_ON_INIT` macro.

Reply:
when do you expect that change to happen?